crates/astria-sequencer/src/app/tests_execute_transaction.rs

```diff
@@ -1100,3 +1109,234 @@ async fn transaction_execution_records_fee_event() {
+    let mut state_tx = StateDelta::new(app.state.clone());
+    state_tx.put_transfer_base_fee(1).unwrap();
+    state_tx.put_bridge_lock_byte_cost_multiplier(1);
+    state_tx.put_bridge_account_rollup_id(bridge_address, &rollup_id);
+    state_tx
+        .put_bridge_account_ibc_asset(bridge_address, nria())
+        .unwrap();
+    app.apply(state_tx);
+
+    let actions = vec![
+        BridgeLockAction {
+            to: bridge_address,
+            amount: 1,
+            asset: nria().into(),
+            fee_asset: nria().into(),
+            destination_chain_address: rollup_id.to_string(),
+        }
+        .into(),
+    ];
+
+    let tx = UnsignedTransaction {
+        params: TransactionParams::builder()
+            .nonce(0)
+            .chain_id("test")
+            .build(),
+        actions,
+    };
+    let signed_tx = Arc::new(tx.into_signed(&alice));
+    app.execute_transaction(signed_tx).await.unwrap();
+
+    let test_deposit = Deposit::new(
+        bridge_address,
+        rollup_id,
+        1,
+        nria().into(),
+        rollup_id.to_string(),
+    );
+
+    let total_block_fees: u128 = app
+        .state
+        .get_block_fees()
+        .await
+        .expect("failed to get block fees")
+        .into_iter()
+        .map(|(_, fee)| fee)
+        .sum();
+    let expected_fees = 1 + get_deposit_byte_len(&test_deposit);
+    assert_eq!(total_block_fees, expected_fees);
+}
+
+#[tokio::test]
+async fn ensure_correct_block_fees_bridge_sudo_change() {
+    let alice = get_alice_signing_key();
+    let alice_address = astria_address(&alice.address_bytes());
+    let bridge = get_bridge_signing_key();
+    let bridge_address = astria_address(&bridge.address_bytes());
+
+    let mut app = initialize_app(None, vec![]).await;
+    let mut state_tx = StateDelta::new(app.state.clone());
+    state_tx.put_bridge_sudo_change_base_fee(1);
+    state_tx.put_bridge_account_sudo_address(bridge_address, alice_address);
+    state_tx
+        .increase_balance(bridge_address, nria(), 1)
+        .await
+        .unwrap();
+    app.apply(state_tx);
+
+    let actions = vec![
+        BridgeSudoChangeAction {
+            bridge_address,
+            new_sudo_address: None,
+            new_withdrawer_address: None,
+            fee_asset: nria().into(),
+        }
+        .into(),
+    ];
+
+    let tx = UnsignedTransaction {
+        params: TransactionParams::builder()
+            .nonce(0)
+            .chain_id("test")
+            .build(),
+        actions,
+    };
+    let signed_tx = Arc::new(tx.into_signed(&alice));
+    app.execute_transaction(signed_tx).await.unwrap();
+
+    let total_block_fees: u128 = app
+        .state
+        .get_block_fees()
+        .await
+        .expect("failed to get block fees")
+        .into_iter()
+        .map(|(_, fee)| fee)
+        .sum();
+    assert_eq!(total_block_fees, 1);
+}
+
+// TODO: Add test to ensure correct block fees for ICS20 withdrawal
```

Comment:
will you add that here?

Reply:
I attempted to do so but ran into a lot of difficulty setting up the IBC channel/ports, which is required for `check_and_execute()`. It looks like Janis ran into the same issue in #1345 because Penumbra doesn't make the `StateWriteExt` traits public. I may be missing something, though. If you have any suggestions it would be greatly appreciated! Also, thanks for the quick review and welcome back!

Reply:
Yeah, gotta make a PR against penumbra.

Reply:
If we are leaving a TODO in after merge, create a follow up issue and link in code please :) ie

``` 
// TODO(https://github.com/astriaorg/astria/issues/<issue num>): add test for correct block fees
```